contracts/dao/ListaRevenueDistributor.sol

```diff
@@ -0,0 +1,139 @@
+/**
+  * @title ListaRevenueDistributor
+  * @dev lista revenue tokens distributor
+  * @dev all tokens revenue (including Lista Token) from Lista project contracts will be sent to ListaRevenueDistributor
+  * @dev in ListaRevenueDistributor, non-ListaToken revenue will be distributed to autoBuybackAddress and revenueWalletAddress according to distributeRate
+  * @dev the distributeRate part of ListaToken revenue will be sent to listaDistributeToAddress instead of autoBuybackAddress
+  */
+contract ListaRevenueDistributor is Initializable, AccessControlUpgradeable {
+
+    using SafeERC20 for IERC20;
+
+    event RevenueDistributed(address indexed token, uint256 amount0, uint256 amount1);
+
+    event AddressChanged(uint128 addressType, address newAddress);
+
+    event RateChanged(uint128 rate);
+
+    bytes32 public constant MANAGER = keccak256("MANAGER");
+
+    uint128 public constant RATE_DENOMINATOR = 1e18;
+
+    // distribute rate of revenue to autoBuybackAddress/listaDistributeToAddress
+    uint128 public distributeRate;
+
+    address public listaTokenAddress;
+
+    address public autoBuybackAddress;
+
+    address public revenueWalletAddress;
+
+    // distributeRate part of lista token revenue will be sent to this address instead of autoBuybackAddress
+    address public listaDistributeToAddress;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(
+        address _admin,
+        address _manager,
+        address _listaTokenAddress,
+        address _autoBuybackAddress,
+        address _revenueWalletAddress,
+        address _listaDistributeToAddress,
+        uint128 _distributeRate
+    ) public initializer {
+        require(_admin != address(0), "admin is the zero address");
+        require(_manager != address(0), "manager is the zero address");
+        require(_listaTokenAddress != address(0), "listaTokenAddress is the zero address");
+        require(_autoBuybackAddress != address(0), "autoBuybackAddress is the zero address");
+        require(_revenueWalletAddress != address(0), "revenueWalletAddress is the zero address");
+        require(_listaDistributeToAddress != address(0), "listaDistributeToAddress is the zero address");
+        require(_distributeRate <= 1e18, "too big rate number");
+
+        __AccessControl_init();
+        _setupRole(DEFAULT_ADMIN_ROLE, _admin);
+        _setupRole(MANAGER, _manager);
+
+        listaTokenAddress = _listaTokenAddress;
+        autoBuybackAddress = _autoBuybackAddress;
+        revenueWalletAddress = _revenueWalletAddress;
+        listaDistributeToAddress = _listaDistributeToAddress;
+        distributeRate = _distributeRate;
+    }
+
+    /**
+     * @dev distribute tokens to autoBuybackAddress and revenueWalletAddress according to distributeRate
+     * @param tokens, tokens to distribute
+     */
+    function distributeTokens(address[] memory tokens) external onlyRole(MANAGER) {
```

Comment:
Add an token whitelist in contract for checking  whether the tokens exist in token whitelist